compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -637,6 +643,42 @@ impl<'a, 'tcx> Decodable<DecodeContext<'a, 'tcx>> for Span {
     }
 }
 
+impl<'a, 'tcx> Decodable<DecodeContext<'a, 'tcx>> for Symbol {
+    fn decode(d: &mut DecodeContext<'a, 'tcx>) -> Self {
+        let tag = d.read_u8();
+
+        match tag {
+            SYMBOL_STR => {
+                let s = d.read_str();
```

Comment:
Can we cache in this case too?

Reply:
As i noted in other comment: https://github.com/rust-lang/rust/pull/98851#discussion_r945820047

So caching at this point probably needless for unique strings (this string will be cached if encountered second time with `SYMBOL_OFFSET ` tag).